comps/dataprep/redis/prepare_doc_redis_on_ray.py

```diff
@@ -0,0 +1,304 @@
+    else:
+        for link in tqdm(link_list, total=len(link_list)):
+            with Timer(f"read document {link}."):
+                data = _parse_html(link)
+            if debug:
+                print("content is: ", data)
+            with Timer(f"ingest document {link} to Redis."):
+                data_to_redis(data)
+    return True
+
+
+@register_microservice(name="opea_service@prepare_doc_redis", endpoint="/v1/dataprep", host="0.0.0.0", port=6007)
+async def ingest_documents(files: List[UploadFile] = File(None), link_list: str = Form(None)):
+    if files and link_list:
+        raise HTTPException(status_code=400, detail="Provide either a file or a string list, not both.")
+
+    if not files and not link_list:
+        raise HTTPException(status_code=400, detail="Must provide either a file or a string list.")
+
+    saved_path_list = []
+    if files:
+        try:
+            if not isinstance(files, list):
+                files = [files]
+            upload_folder = "./uploaded_files/"
+            if not os.path.exists(upload_folder):
+                Path(upload_folder).mkdir(parents=True, exist_ok=True)
+
+            # TODO: use ray to parallelize the file saving
+            for file in files:
+                save_path = upload_folder + file.filename
+                await save_file_to_local_disk(save_path, file)
+                saved_path_list.append(DocPath(path=save_path))
+
+            if len(saved_path_list) <= 10:
+                enable_ray = False
+            else:
+                enable_ray = True
+            prepare_env(enable_ray=enable_ray)
+            num_cpus = get_max_cpus(len(saved_path_list))
+            print(f"per task num_cpus: {num_cpus}")
+            ingest_data_to_redis(saved_path_list, enable_ray=enable_ray, num_cpus=num_cpus)
+            return {"status": 200, "message": "Data preparation succeeded"}
+        except Exception as e:
+            raise HTTPException(status_code=400, detail=f"An error occurred: {e}")
+
+    if link_list:
+        try:
+            link_list = json.loads(link_list)  # Parse JSON string to list
+            if not isinstance(link_list, list):
+                raise HTTPException(status_code=400, detail="link_list should be a list.")
+            if len(link_list) <= 10:
+                enable_ray = False
+            else:
+                enable_ray = True
+            prepare_env(enable_ray=enable_ray)
+            num_cpus = get_max_cpus(len(link_list))
+            print(f"per task num_cpus: {num_cpus}")
+            ingest_link_to_redis(link_list, enable_ray=enable_ray, num_cpus=num_cpus)
+            return {"status": 200, "message": "Data preparation succeeded"}
+        except json.JSONDecodeError:
+            raise HTTPException(status_code=400, detail="Invalid JSON format for link_list.")
+        except Exception as e:
+            raise HTTPException(status_code=400, detail=f"An error occurred: {e}")
```

Comment:
@ftian1 , all error including timeout will be returned through 400 for python side. For Ray side, since this is for large data processing, I can return which files didn't been correctly processed in success "200"